app/src/main/java/com/github/se/eventradar/viewmodel/EventsOverviewViewModel.kt

```diff
@@ -40,38 +46,63 @@ constructor(
     }
   }
 
-  fun getUpcomingEvents(uid: String) {
+  fun getUpcomingEvents() {
     viewModelScope.launch {
-      when (val userResponse = userRepository.getUser(uid)) {
-        is Resource.Success -> {
-          val user = userResponse.data!!
-          val attendeeList = user.eventsAttendeeSet.toList()
-          if (attendeeList.isNotEmpty()) {
-            when (val events = eventRepository.getEventsByIds(attendeeList)) {
-              is Resource.Success -> {
-                _uiState.value =
-                    _uiState.value.copy(
-                        eventList =
-                            EventList(
-                                events.data, events.data, _uiState.value.eventList.selectedEvent))
-              }
-              is Resource.Failure -> {
-                Log.d("EventsOverviewViewModel", "Error getting events for $uid")
-                _uiState.value =
-                    _uiState.value.copy(eventList = EventList(emptyList(), emptyList(), null))
-              }
-            }
-          } else {
+      userRepository.getCurrentUserId().let { userIdResource ->
+        when (userIdResource) {
+          is Resource.Success -> {
+            val uid = userIdResource.data
+            getUserEvents(uid)
+          }
+          is Resource.Failure -> {
+            Log.d(
+                "EventsOverviewViewModel",
+                "Error fetching user ID: ${userIdResource.throwable.message}")
             _uiState.value =
                 _uiState.value.copy(eventList = EventList(emptyList(), emptyList(), null))
           }
         }
-        is Resource.Failure -> {
-          Log.d("EventsOverviewViewModel", "Error fetching user document")
+      }
+    }
+  }
+
+  private suspend fun getUserEvents(uid: String) {
```

Comment:
small nit: i know that we separated hosting viewmodel and eventsOverviewViewModel, but maybe getUserUpcomingEvents is more clear so it does not get confused with getUserHostedEvents.